src/components/MapView/DateSelector/index.tsx

```diff
@@ -43,38 +51,148 @@ function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
   const dispatch = useDispatch();
   const { startDate: stateStartDate } = useSelector(dateRangeSelector);
 
-  const stateStartDateYear = moment(stateStartDate).year();
   const [selectedDate, setSelectedDate] = useState(moment(stateStartDate));
-  const selectedMonth = selectedDate.month();
+  const [dateRange, setDateRange] = useState([
```

Comment:
Can we type this? In the generic arguments `<>`. While its inferred, its a little complex to be be left to TS to guess.

Reply:
is it correct?
`const [dateRange, setDateRange] = useState<DateRangeProps[]>([
    {
      value: 0,
      label: '',
      month: '',
      isFirstDay: false,
    },
  ]);
`
with DateRangeProps as
`type DateRangeProps = {
  value: number;
  label: string;
  month: string;
  isFirstDay: boolean;
};
`